src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -331,8 +333,20 @@ public void close(Collection<TopicPartition> partitions) {
               topicPartitionChannel == null ? null : topicPartitionChannel.getChannelName(),
               topicPartition.topic(),
               topicPartition.partition());
+          partitionsToChannel.remove(partitionChannelKey);
         });
-    partitionsToChannel.clear();
+    List<String> partitionsNotClosed =
+        partitionsToChannel.keySet().stream()
+            .filter(
+                key ->
+                    partitions.stream()
+                        .noneMatch(
+                            tp -> key.equals(partitionChannelKey(tp.topic(), tp.partition()))))
+            .collect(Collectors.toList());
```

Comment:
Why can't we just print the partitionsToChannel since the matched elements should be all removed?

Reply:
yikes.. let me do that! haha